go/service/gregor.go

```diff
@@ -91,11 +92,109 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
 func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
 	g.G().Log.Debug("gregor handler: broadcast: %+v", m)
 
+	ibm := m.ToInBandMessage()
+	if ibm != nil {
+		return g.handleInBandMessage(ctx, ibm)
+	}
+
 	obm := m.ToOutOfBandMessage()
-	if obm == nil {
-		return fmt.Errorf("gregor handler: out-of-band message nil")
+	if obm != nil {
+		return g.handleOutOfBandMessage(ctx, obm)
+	}
+
+	return fmt.Errorf("gregor handler: both in-band and out-of-band message nil")
+}
+
+func (g *gregorHandler) handleInBandMessage(ctx context.Context, ibm gregor.InBandMessage) error {
+	g.G().Log.Debug("gregor handler: handleInBand: %+v", ibm)
+
+	sync := ibm.ToStateSyncMessage()
+	if sync != nil {
+		g.G().Log.Debug("gregor handler: state sync message")
```

Comment:
Are we certain we will never have a message that has both sync and update? I guess we are?